shell/platform/android/io/flutter/embedding/android/FlutterActivity.java

```diff
@@ -404,11 +404,11 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
 
     super.onCreate(savedInstanceState);
 
-    lifecycle.handleLifecycleEvent(Lifecycle.Event.ON_CREATE);
-
     delegate = new FlutterActivityAndFragmentDelegate(this);
     delegate.onAttach(this);
-    delegate.onActivityCreated(savedInstanceState);
+    delegate.onRestoreInstanceState(savedInstanceState);
```

Comment:
Sounds reasonable. Could you add a test that would have otherwise failed without this in FlutterActivityTest or FlutterAndroidComponentTest? Our general docs for engine testing is at https://github.com/flutter/flutter/wiki/Testing-the-engine.

Reply:
Done.